--- dwm-6.0/config.def.h.orig	2015-04-16 09:36:08.604336958 -0400
+++ dwm-6.0/config.def.h	2015-04-16 09:26:05.647771882 -0400
@@ -43,7 +43,7 @@
 };
 
 /* key definitions */
-#define MODKEY Mod1Mask
+#define MODKEY Mod4Mask
 #define TAGKEYS(KEY,TAG) \
 	{ MODKEY,                       KEY,      view,           {.ui = 1 << TAG} }, \
 	{ MODKEY|ControlMask,           KEY,      toggleview,     {.ui = 1 << TAG} }, \
@@ -55,7 +55,10 @@
 
 /* commands */
 static const char *dmenucmd[] = { "dmenu_run", "-fn", font, "-nb", colors[0][ColBG], "-nf", colors[0][ColFG],"-sb", colors[1][ColBG], "-sf", colors[1][ColFG], NULL };
-static const char *termcmd[]  = { "uxterm", NULL };
+static const char *termcmd[]  = { "urxvt", NULL };
+static const char *vol_up[]      = { "pulseaudio-ctl", "up", NULL};
+static const char *vol_down[]    = { "pulseaudio-ctl", "down", NULL};
+static const char *mute_toggle[] = { "pulseaudio-ctl", "mute", NULL};
 
 static Key keys[] = {
 	/* modifier                     key        function        argument */
