--- dwm-6.1/config.def.h.orig	2015-12-02 22:44:43.039134325 -0500
+++ dwm-6.1/config.def.h	2015-12-02 22:44:58.045431378 -0500
@@ -2,9 +2,9 @@
 
 /* appearance */
 static const char *fonts[] = {
-	"monospace:size=10"
+	"fixed-10:style=semicondensed"
 };
-static const char dmenufont[]       = "monospace:size=10";
+static const char dmenufont[]       = "fixed-10:style=semicondensed";
 
 #define NUMCOLORS         8
 static const char colors[NUMCOLORS][3][8] = { // eww
