--- dwm-6.0/config.def.h.orig	2015-04-16 09:38:54.451857156 -0400
+++ dwm-6.0/config.def.h	2015-04-16 09:26:05.647771882 -0400
@@ -22,16 +22,18 @@
 static const Bool topbar            = True;     /* False means bottom bar */
 
 /* tagging */
-static const char *tags[] = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };
+static const char *tags[] = { "one", "two", "three", "four", "five", "six", "seven", "eight", "nine" };
 
 static const Rule rules[] = {
 	/* xprop(1):
 	 *	WM_CLASS(STRING) = instance, class
 	 *	WM_NAME(STRING) = title
 	 */
 	/* class      instance    title       tags mask     isfloating   monitor */
 	{ "Gimp",     NULL,       NULL,       0,            1,           -1 },
-	{ "Firefox",  NULL,       NULL,       1 << 8,       0,           -1 },
+	{ "Firefox",  NULL,       NULL,       1 << 0,       0,           -1 },
+	{ "Chromium", NULL,       NULL,       1 << 0,       0,           -1 },
+	{ "Emacs",    NULL,       NULL,       1 << 1,       0,           -1 },
 };
 
 /* layout(s) */
